gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientReadChannel.java

```diff
@@ -582,6 +586,10 @@ private BlobSourceOption[] generateReadOptions(BlobId blobId) {
         blobReadOptions.add(
             BlobSourceOption.decryptionKey(storageOptions.getEncryptionKey().value()));
       }
+      if (requesterShouldPay.apply((blobId.getBucket()))) {
```

Comment:
Why can't we have this as a boolean value per GCSClientReadChannel? i.e. what is the value of recomputing this for reach GCS ReadChannel?